src/components/H/H.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import Box, { BoxProps } from '../Box';
+
+const DEFAULT_HEADING_LEVEL = 2;
+export const HeadingLevelContext = React.createContext(DEFAULT_HEADING_LEVEL);
+
+type HTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+export type HProps = BoxProps<HTag>;
+
+/**
+ * Provides and manages `h` tags within a document context. The underlying heading level rendered
+ * can vary depending on _where_ this component is in the component tree.
+ *
+ * This allows for simple scaffolding of a document with accessible heading levels while
+ * also allowing components to create markup in a vacuum.
+ */
+export const H: React.FC<HProps> = React.forwardRef<HTMLHeadingElement, HProps>(function H(
```

Comment:
question: why don't we replace our existing `Heading` with this? I don't see any reasson to have both:

- `H`
- `Heading`

As a dev, I'm a bit confused by the presence of both. Are there any downsides (apart from the context reading which is fine) that you see?

P.S. We'll have to handle `level` being `undefined` in that case, but that's easy to fix